variables.tf

```diff
@@ -137,12 +143,27 @@ variable "site_config" {
   default = {}
 }
 
-variable "identity_ids" {
-  description = <<EOT
-    Specifies a list of user managed identity ids to be assigned.
-  EOT
-  type        = list(string)
+variable "identity" {
+  description = "(Optional) An identity block."
+  type = object({
+    type         = string
+    identity_ids = optional(list(string))
+  })
+  default = null
```

Comment:
This fails when identity=null

```│ Error: Invalid index
│
│   on ../../outputs.tf line 30, in output "principal_id":
│   30:   value       = lower(var.os_type) == "linux" ? azurerm_linux_function_app.func[0].identity[0].principal_id : azurerm_windows_function_app.windows_func[0].identity[0].principal_id
│     ├────────────────
│     │ azurerm_linux_function_app.func[0].identity is empty list of object
│
│ The given key does not identify an element in this collection value: the collection has no elements.

```

Reply:
I have corrected this and accounted for this scenario via a test.